src/os/bluestore/BlueStore.cc

```diff
@@ -8930,6 +8930,12 @@ int BlueStore::_split_collection(TransContext *txc,
   c->onode_map.clear();
   d->onode_map.clear();
 
+  // We have an awkward race here: previous pipelinex transactions may
+  // still reference blobs and their shared_blobs.  They will be flushed
+  // shortly by _osr_reap_done, but it's awkward to block for that (and
+  // a waste of time).  Instead, explicitly remove them from the shared blob
+  // map.
+  c->shared_blob_set.violently_clear();
   assert(c->shared_blob_set.empty());
   assert(d->shared_blob_set.empty());
```

Comment:
Not sure if this makes much sense but is it possible that destination collection struggles (or will struggle one day) with the same issue? Does it worth to clear its shared_blob_set the same way? Just in case...